### PR TITLE
Add workflow for hz operator image release

### PR DESCRIPTION
**Pull request by mtyazici**
Added workflow for Hazelcast-operator and Hazelcast-enterprise-operator Docker Hub release. It does the following:
- Install all dependencies to complete the process
- Downloads the Hazelcast Helm chart and overwrites some fields in `statefulset.yml` and `values.yml`
- Generates operator files, then the operator image
- Updates the values in files that are used for deployment test in Kubernetes.
- Sets up a Kubernetes in Docker(Kind) environment
- Deploys operator image into Kind, and waits for it to be `ready`
- Deploys Hazelcast members and a management center, and waits for all of them to be `ready`
- Checks the cluster size by looking at the log of the last Hazelcast member
- Checks the management center logs to verify members can talk to the management center
- Creates Dockerhub operator bundle and uploads it as an artifact
- Pushes the operator image to Docker Hub
- Updates the versions the hazelcast-operator repo and creates a PR for the changes
- Pushes new operatorhub bundle to `DevopsHelm/community-operators` fork, creates a PR for  `operator-framework /
community-operators`

### CODE REVIEW

File: .github/workflows/hz-dockerhub-release.yml [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+                kubectl get pods
+                echo ""
+                kubectl logs -l app.kubernetes.io/name=${OPERATOR_NAME}
+                exit 1
+              fi
+            else
+              echo "Hazelcast operator pod is ready!"
+              break
+            fi
+          done
```
Comment (hasancelik) [inline .github/workflows/hz-dockerhub-release.yml:216]:
```suggestion
kubectl wait --for=condition=Ready --timeout=60s pod -l app.kubernetes.io/name=${OPERATOR_NAME}
```

File: .github/workflows/hz-dockerhub-release.yml [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+                kubectl get pods
+                echo ""
+                kubectl logs hz-${NAME}-2
+                exit 1
+              fi
+            else
+              echo "hz-${NAME}-2 is ready!"
+              break
+            fi
+          done
```
Comment (hasancelik) [inline .github/workflows/hz-dockerhub-release.yml:237]:
```suggestion
 kubectl wait --for=condition=Ready --timeout=60s pod -l _LABEL_OF_HZ_POD_=${OPERATOR_NAME}
```
You can use one of below labels as a `_LABEL_OF_HZ_POD_`:
https://github.com/hazelcast/hazelcast-operator/blob/master/hazelcast-operator/hazelcast.yaml#L6-L8

File: .github/workflows/hz-dockerhub-release.yml [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+                exit 1
+              fi
+            else
+              echo "Hazelcast operator pod is ready!"
+              break
+            fi
+          done
+
+          kubectl apply -f hazelcast.yaml
+          kubectl logs -l app.kubernetes.io/name=${OPERATOR_NAME}
```
Comment (hasancelik) [inline .github/workflows/hz-dockerhub-release.yml:219]: Why logs are printed to terminal?

File: .github/workflows/hz-dockerhub-release.yml [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+                kubectl logs hz-${NAME}-2
+                exit 1 
+              fi
+            fi
+          done
+
+
+      - name: Copy operatorhub-bundle from the repo
+        run: |
+          cp -r ./operator-repo/.github/operator-release-files/operatorhub-bundle \
```
Comment (hasancelik) [inline .github/workflows/hz-dockerhub-release.yml:261]:
```suggestion
          cp -r ./operator-repo/.github/operator-release-files/operatorhub-bundle ./
```

**Review by hasancelik — CHANGES_REQUESTED** [review]
Great job @mtyazici 👏 I have added some comments, please check them.

File: .github/workflows/hz-dockerhub-release.yml [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+      - name: Install oc
+        run: |
+          set -x
+          curl -L -o ./oc.tar.gz "https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/linux/oc.tar.gz"
+          tar xvzf oc.tar.gz
+          rm oc.tar.gz
+          sudo mv oc /oc
+          /oc version
+        
+      - name: Check for utility packages
```
[the remark this answers is not in the document]
Comment (mtyazici) [reply .github/workflows/hz-dockerhub-release.yml:65]: I will remove it.

File: .github/workflows/hz-dockerhub-release.yml [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+            overrideValues:
+              image.repositoryOverride: \${HAZELCAST_IMAGE_REPOSITORY}
+              image.tagOverride: \${HAZELCAST_IMAGE_TAG}
+              mancenter.image.repositoryOverride: \${MC_IMAGE_REPOSITORY}
+              mancenter.image.tagOverride: \${MC_IMAGE_TAG}
+          EOL
+
+          echo "Building the operator image ${OPERATOR_IMAGE}"
+          /operator-sdk build ${OPERATOR_IMAGE}
+
```
[the remark this answers is not in the document]
Comment (mtyazici) [reply .github/workflows/hz-dockerhub-release.yml:160]: I was going to keep it but I didn't see the use of `deploy/operator.yaml` anywhere, can you explain a bit more?

File: .github/workflows/hz-dockerhub-release.yml (hunk 1 above)
Comment (mtyazici) [reply .github/workflows/hz-dockerhub-release.yml:216]: Oh this is a very hand command, I was not aware of that, thanks!

File: .github/workflows/hz-dockerhub-release.yml (hunk 3 above)
Comment (mtyazici) [reply .github/workflows/hz-dockerhub-release.yml:219]: When any problem occurs we will look at the workflow run, I thought logs woul be helpful inspecting the problem

**Comment by mtyazici** [comment]
New workflow run outpus is here: https://github.com/mtyazici/hazelcast-operator/actions/runs/382736551. It fails because I didn't give docker-hub credentials. However it builds everything correctly.

**Review by hasancelik — APPROVED** [review]
LGTM 👍 💯

**Pull request merged**